File: app/template/src/supervisors/supervisor-template.c
#include <stdbool.h>
#include <stdlib.h>

#include "../event_handler/events.h"
#include "%$%{sup_name_source}.h"

// Supervisor specific instances
// alphabet create
%$%{alphabet_create_source}
// Alphabet init
%$%{alphabet_init_source}
// states create
%$%{state_create_source}
// transitions create and init
%$%{transitions_list_source}
// states init
%$%{state_init_source}
// Supervisor create
%$%{supervisor_create_source}

File: app_python/template/Supervisor/supervisors/supervisor-template.py
from .. import Events, State, Supervisor

# Create states
%$%{states}

# Create transitions
%$%{transitions}

# create state list
state_list = []
%$%{state_list}

# create event list
alphabet = []
%$%{alphabet}

# create supervisor
%$%supervisor_name = Supervisor("%$%supervisor_name", state_list, alphabet)